File: Algorithm-CPP/24_JollyJumpers.cpp
#include <iostream>
#include <math.h>
using namespace std;

int main() {
    int n;
    cin >> n;

    int arr[n], jumpers[n];
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    for(int i = 0; i < n; i++) {
        jumpers[i] = 0;
    }

    for(int i = 0; i < n - 1; i++) {
        int result = abs(arr[i+1] - arr[i]);
        if(result < n && result > 0) {
            jumpers[result] = 1;
        } else {
            cout << "NO" << endl;
            return 0;
        }
    }

    for(int i = 1; i < n; i++) {
        if(jumpers[i] == 0) {
            cout << "NO" << endl;
            return 0;
        }
    }

    cout << "YES" << endl;

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/24_JollyJumpers.swift
//
//  24_JollyJumpers.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/04.
//

import Foundation

func solution24() {
    let n = Int(readLine()!)!
    let nums = readLine()!.split(separator: " ").map { Int($0)! }
    var jumpers = Array(repeating: 0, count: n)

    for i in stride(from: 0, to: n-1, by: 1) {
        let result = abs(nums[i+1] - nums[i])
        if result > 0 && result < n {
            jumpers[result] = 1
        } else {
            print("NO")
            return
        }
    }

    for i in 1..<n {
        if jumpers[i] == 0 {
            print("NO")
            return
        }
    }

    print("YES")
}
